src/selectors/index.ts

```diff
@@ -19,9 +18,9 @@ export const getQuery = createSelector(
 );
 
 export const getMainResult = createSelector(
-  getQuery, getSchema, getData,
-  (query: Query, schema: Schema): SpecQueryModelGroup => {
-    return recommend(query, schema).result;
+  getRecommends,
```

Comment:
If we rename `getRecommends` to `getMainResult`, I think we don't really need this composite selector anymore.